idioms/return-moved-arg-on-error.md

```diff
@@ -0,0 +1,62 @@
+# A succinct name for the pattern
+
+## Description
+
+If a fallible function consumes an argument, return that argument back inside
+an error.
+
+## Example
+
+```rust
+pub fn send(value: String) -> Result<(), SendError> {
+    // Simulate non-deterministic fallible action.
```

Comment:
```suggestion
    println!("using {value} in a meaningful way");
    // Simulate non-deterministic fallible action.
```
We could write something like this to make it explicit that the function is using the value. Actually, it would be great if we could think of an example where the value is used for real.